core/o11y/src/lib.rs

```diff
@@ -41,17 +43,25 @@ macro_rules! io_trace {
     ($($fields:tt)*) => {};
 }
 
-static LOG_LAYER_RELOAD_HANDLE: OnceCell<
+static LOG_LAYER_RELOAD_HANDLE: OnceCell<Handle<EnvFilter, Registry>> = OnceCell::new();
+
+static OTLP_LAYER_RELOAD_HANDLE: OnceCell<
     Handle<
-        Filtered<
-            tracing_subscriber::fmt::Layer<Registry, DefaultFields, Format, NonBlocking>,
-            EnvFilter,
+        LevelFilter,
+        Layered<
+            Filtered<
+                tracing_subscriber::fmt::Layer<Registry, DefaultFields, Format, NonBlocking>,
+                reload::Layer<EnvFilter, Registry>,
```

Comment:
Is the `EnvFilter` actually useful for OTLP traces? For logging I see the benefit in providing granular control. Te decision of what you want to see in your logs must be made up-front (otherwise the logs end up heavily cluttered, possibly without an obvious way to filter it out later). With OTLP, usually there’s a mechanism to filter events and spans by field/etc in some sort of viewer (e.g. grafana) after the fact.

My intuition is that `EnvFilter` has a non-trivial overhead here, and so if we can stick with just a plain level, I suspect we’d be better off.

Reply:
This type signature is a subscriber with the logging layer. `EnvFilter` refers to the filter used by the log layer, not the opentelemetry layer.
This should be clarified with the new type alias.